handsontable/src/stylesManager.js

```diff
@@ -3,31 +3,197 @@
  */
 export class StylesManager {
   /**
-   * The Handsontable instance.
+   * The Handsontable instance's root element.
+   *
+   * @type {HTMLElement}
+   */
+  #rootElement;
+
+  /**
+   * The computed style of the root element.
+   *
+   * @type {CSSStyleDeclaration}
+   * @private
    */
-  #hot;
+  #rootComputedStyle;
+
+  /**
+   * `true` if the legacy theme is enabled, `false` otherwise.
+   *
+   * @type {boolean}
+   */
+  static #legacyTheme = false;
+
   /**
    * Keeps a value of cell vertical padding css variable.
    *
    * @type {number}
    */
-  cellVerticalPadding;
+  static #cellVerticalPadding;
+
+  /**
+   * Keeps a value of cell horizontal padding css variable.
+   *
+   * @type {number}
+   */
+  static #cellHorizontalPadding;
+
+  /**
+   * `box-sizing` value of the cell.
+   *
+   * @type {string}
+   */
+  static #cellBoxSizing;
+
+  /**
+   * Keeps a value of cell bottom border width css variable.
+   *
+   * @type {number}
+   */
+  static #cellBottomBorderWidth;
+
+  /**
+   * Keeps a value of cell top border width css variable.
+   *
+   * @type {number}
+   */
+  static #cellTopBorderWidth;
+
+  /**
+   * Keeps a value of cell line height css variable.
+   *
+   * @type {number}
+   */
+  static #cellLineHeight;
+
+  /**
+   * Creates an instance of StylesManager.
+   *
+   * @param {HTMLElement} rootElement - The root element of the Handsontable instance.
+   */
+  constructor(rootElement) {
+    this.#rootElement = rootElement;
+    this.#rootComputedStyle = getComputedStyle(this.#rootElement);
+
+    if (this.#rootComputedStyle?.getPropertyValue('--ht-line-height') === '') {
+      StylesManager.#legacyTheme = true;
+
+      return;
+    }
 
-  constructor(hotInstance) {
-    this.#hot = hotInstance;
+    StylesManager.#cellVerticalPadding = this.#getParsedCSSValue('--ht-cell-vertical-padding');
+    StylesManager.#cellHorizontalPadding = this.#getParsedCSSValue('--ht-cell-horizontal-padding');
+    StylesManager.#cellLineHeight = this.#getParsedCSSValue('--ht-line-height');
 
-    this.cellVerticalPadding = parseInt(
-      getComputedStyle(this.#hot.rootElement)?.getPropertyValue('--ht-cell-vertical-padding') || 0,
-      10
+    // Some of the properties are not stored in the CSS variables, so we need to get them from the computed style.
+    this.#getCSSValuesFromTempElement(
+      { type: 'div', className: 'handsontable' },
+      { type: 'td', className: '' },
+      (computedStyle) => {
+        StylesManager.#cellBoxSizing = computedStyle['box-sizing'];
+        StylesManager.#cellTopBorderWidth = parseInt(computedStyle['border-top-width'], 10);
+        StylesManager.#cellBottomBorderWidth = parseInt(computedStyle['border-bottom-width'], 10);
+      },
     );
```

Comment:
I have the impression that this method is made to be generic and it seems to me that it is not necessary (it is not used anywhere else). I would suggest something like this (pseudocode):
```js
this.#applyDynamicCSSValues(); 
```
and the method could be simplified to something like:
```js
  #applyDynamicCSSValues() {
    const parentElement = doc.createElement('div');
    parentElement.style.display = 'none';
    parentElement.className = 'handsontable';

    const childElement = doc.createElement('div');

    doc.body.appendChild(parentElement);
    parentElement.appendChild(childElement);

    const computedStyle = getComputedStyle(childElement);

    StylesManager.#cellBoxSizing = computedStyle['box-sizing'];
    StylesManager.#cellTopBorderWidth = parseInt(computedStyle['border-top-width'], 10);
    StylesManager.#cellBottomBorderWidth = parseInt(computedStyle['border-bottom-width'], 10);

    this.#rootElement.body.removeChild(parentElement);
  }
```